src/App.tsx

```diff
@@ -1,9 +1,21 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const visibleMovies = moviesFromServer.filter(movie => {
+    const movieTitleInLowerCase = movie.title.toLowerCase();
+    const queryInLowerCase = query.toLowerCase();
+    const descriptionInLowerCase = movie.description.toLowerCase();
+    const rightTitle = movieTitleInLowerCase.includes(queryInLowerCase);
+    const rightDescription = descriptionInLowerCase.includes(queryInLowerCase);
+
+    return rightTitle || rightDescription;
+  });
```

Comment:
So that `filter` is performed each time the component re-renders?

Reply:
I'm not sure that I got it clearly...and how can I make it.